Parkinson/Startup/APHAppDelegate.m

```diff
@@ -209,28 +325,379 @@ - (NSArray *)allSetTextBlocks
+            {
+                // This is really important to remember that we are creating as many user defaults as there are healthkit permissions here.
+                NSString*                               uniqueAnchorDateName    = [NSString stringWithFormat:@"APCHealthKit%@AnchorDate", dataType];
+                APCHealthKitBackgroundDataCollector*    collector               = nil;
+                
+                //If the HKObjectType is a HKWorkoutType then set a different receiver/data sink.
+                if ([sampleType isKindOfClass:[HKWorkoutType class]])
+                {
+                    collector = [[APCHealthKitBackgroundDataCollector alloc] initWithIdentifier:sampleType.identifier
+                                                                                     sampleType:sampleType anchorName:uniqueAnchorDateName
+                                                                               launchDateAnchor:LaunchDate
+                                                                                    healthStore:self.dataSubstrate.healthStore];
+                    [collector setReceiver:workoutReceiver];
+                    [collector setDelegate:workoutReceiver];
+                }
+                else if ([sampleType isKindOfClass:[HKCategoryType class]])
+                {
+                    collector = [[APCHealthKitBackgroundDataCollector alloc] initWithIdentifier:sampleType.identifier
+                                                                                     sampleType:sampleType anchorName:uniqueAnchorDateName
+                                                                               launchDateAnchor:LaunchDate
+                                                                                    healthStore:self.dataSubstrate.healthStore];
+                    [collector setReceiver:sleepReceiver];
+                    [collector setDelegate:sleepReceiver];
+                }
+                else
+                {
+                    NSDictionary* hkUnitKeysAndValues = [self researcherSpecifiedUnits];
+                    
+                    collector = [[APCHealthKitBackgroundDataCollector alloc] initWithQuantityTypeIdentifier:sampleType.identifier
+                                                                                                 sampleType:sampleType anchorName:uniqueAnchorDateName
+                                                                                           launchDateAnchor:LaunchDate
+                                                                                                healthStore:self.dataSubstrate.healthStore
+                                                                                                       unit:[hkUnitKeysAndValues objectForKey:sampleType.identifier]];
+                    [collector setReceiver:quantityreceiver];
+                    [collector setDelegate:quantityreceiver];
+                }
+                
```

Comment:
Consolidate `initWithQuantityTypeIdentifier `/`setReceiver `/`setDelegate` by moving them out of these if...else blocks.

e.g.
```
if () {
...
} else if () {
...
} else {
...
}

collector = [[APCHealthKitBackgroundDataCollector alloc] initWithQuantityTypeIdentifier:sampleType.identifier
sampleType:sampleType anchorName:uniqueAnchorDateName
launchDateAnchor:LaunchDate
healthStore:self.dataSubstrate.healthStore
unit:unit];


[collector setReceiver:receiver];
[collector setDelegate:receiver];
```